specifyweb/specify/views.py

```diff
@@ -528,6 +537,59 @@ def update_record(record: models.Model):
 
     # Return http response
     return http.HttpResponse('', status=204)
+
+@app.task(base=LogErrorsTask, bind=True)
+def record_merge_task(self, user_id: int, model_name: str, old_model_ids: List[int], new_model_id: int, merge_id: int,
+                      new_record_dict: Dict[str, Any]=None):
+
+    Collection = getattr(models, 'Collection')
+    Specifyuser = getattr(models, 'Specifyuser')
+    specify_user = Specifyuser.objects.get(id=new_record_dict['specify_user_id'])
+    new_record_info = {
+        'agent_id': new_record_dict['agent_id'],
+        'collection': Collection.objects.get(id=new_record_dict['collection_id']),
+        'specify_user': specify_user,
+        'version': new_record_dict['version'],
+        'new_record_data': new_record_dict['new_record_data']
+    }
+
+    "Run the record merging process as a background task with celery"
+    current = 0
+    total = 1
+
+    # Track the progress of the record merging
+    def progress(cur: int, additional_total: int=0) -> None:
+        nonlocal current, total
+        current += cur
+        total += additional_total
+        if not self.request.called_directly:
+            self.update_state(state='MERGING', meta={'current': current, 'total': total})
+
+    # Create a notification record of the merging process starting
+    Message.objects.create(user=specify_user, content=json.dumps({
+        'type': 'record-merge-started',
+        'name': "Merge_" + model_name + "_" + new_model_id,
+        'task_id': self.id,
+    }))
+
+    # Run the record merging function
+    response = record_merge_fx(model_name, old_model_ids, int(new_model_id), progress, new_record_info)
+
+    # Update the finishing state of the record merging process
+    merge_record = Spmerging.objects.get(id=merge_id)
+    if response.status_code == 204:
+        self.update_state(state='SUCCEEDED', meta={'current': total, 'total': total})
```

Comment:
(optional) Enum for state or mergingstatus? https://docs.python.org/3/library/enum.html

Reply:
then, you would be able to reuse that enum on line 764

https://stackoverflow.com/questions/29503339/how-to-get-all-values-from-python-enum-class